projects/ion/src/lib/checkbox/checkbox.component.html

```diff
@@ -1,7 +1,15 @@
-<input
-  data-testid="ion-checkbox"
-  class="ion-checkbox"
-  type="checkbox"
-  (click)="changeState()"
-  #checkBox
-/>
+<div class="ion-input-checkbox">
+  <input
+    data-testid="ion-checkbox"
+    class="ion-checkbox"
+    type="checkbox"
+    (click)="changeState()"
+    #checkBox
+  />
+  <label
+    *ngIf="label"
+    [ngClass]="{ labelEnabled: !disabled, labelDisabled: disabled }"
```

Comment:
You don't think here can be worked with just one class?

You can just set the enable style to the label element in CSS file, and just use a specific class to disabled.

`label { cursor: pointer; color: $neutral-7; }`

`.labelDisabled { cursor: not-allowed; color: $neutral-5; }`

And in the HTML you can just pass this:

`[class.labelDisabled]="disabled"`

Reply:
right.